### PR TITLE
doc: fix: Fix links in README.md

### PR DESCRIPTION
**Pull request by oskarwojciski**
Fix links - connected to issue #344

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,98 +13,86 @@ Read our [Contribution Guidelines](CONTRIBUTING.md) before you contribute.
...
+### Math
+* [Gcd](./math/gcd)
+* [Lcm](./math/lcm)
+* [Max](./math/max)
+* [Modular](./math/modular) 
+* [Moser-de Bruijn sequence](./math/moserdebruijnsequence)
+* [Permutation](./math/permutation)
+* [Power](./math/power)
+* [Prime](./math/prime)
+* [Pythagoras](./math/pythagoras)
```
Comment (tjgurwara99) [inline README.md:64]:
```suggestion
* [Gcd](./math/gcd)
    * [Extended](./math/gcd/extended.go)
    * [Recursive](./math/gcd/gcd.go)
    * [Iterative](./math/gcd/gcditerative.go)
* [Lcm](./math/lcm/lcm.go)
* [Max](./math/max/max.go)
* [Modular](./math/modular)
    * [Exponentiation](./math/max/max.go)
    * [Inverse](./math/modular/inverse.go) 
* [Moser-de Bruijn sequence](./math/moserdebruijnsequence/sequence.go)
* [Permutation](./math/permutation/heaps.go)
* [Power](./math/power/fastexponent.go)
* [Prime](./math/prime)
    * [Miller Rabin Primality Test](./math/prime/millerrabinprimalitytest.go)
    * [Naive and Pair Approach](./math/prime/primecheck.go)
    * [Sieve](./math/prime/sieve.go)
* [Pythagoras](./math/pythagoras/pythagoras.go)
```

File: README.md [hunk 2]
```diff
@@ -13,98 +13,86 @@ Read our [Contribution Guidelines](CONTRIBUTING.md) before you contribute.
...
+* [Sieve](./math/prime/sieve.go)
+
+### Other
+* [Max Subarray Sum](./other/maxsubarraysum)
+* [Nested Brackets](./other/nested)
+* [Password Generator](./other/password)
```
Comment (tjgurwara99) [inline README.md:70]:
```suggestion
* [Max Subarray Sum](./other/maxsubarraysum/maxsubarraysum.go)
* [Nested Brackets](./other/nested/nestedbrackets.go)
* [Password Generator](./other/password/generator.go)
```

File: README.md [hunk 3]
```diff
@@ -13,98 +13,86 @@ Read our [Contribution Guidelines](CONTRIBUTING.md) before you contribute.
...
-* [Naive String Search](./strings/naive_string_search/)
+    * [Advanced Aho Corasick](./strings/ahocorasick/advancedahocorasick.go)
+    * [Aho Corasick](./strings/ahocorasick/ahocorasick.go)
 * Single String Matching
-    * [Bom](./strings/single_string_matching/bom/)
-    * [Horspool](./strings/single_string_matching/horspool/)
-    * [Kmp](./strings/single_string_matching/kmp/)
+    * [Bom](./strings/bom)
+    * [Horspool](./strings/horspool)
+    * [Kmp](./strings/kmp)
```
Comment (tjgurwara99) [inline README.md:98]:
```suggestion
    * [Bom](./strings/bom/bom.go)
    * [Horspool](./strings/horspool/horspool.go)
    * [Kmp](./strings/kmp/kmp.go)
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Thanks for doing this. I have added some suggestions for changes. I think it is best to keep directing people to `*.go` files instead of directories as directories can be overwhelming for beginners. Most modern IDE's can go to the definition of functions when you `ctrl/cmd + click` on them so arguably it makes it easier to do that from inside the file.

Also could I kindly ask you to take a look at some other listings here that you have not edited but are directing to a directory instead of `*.go` files.

**Comment by tjgurwara99** [comment]
Re-request review by clicking there when you think this PR ready to be reviewed😄

Screenshot 2021-09-10 at 21 44 59

**Review by tjgurwara99 — APPROVED** [review]

**Review by tjgurwara99 — APPROVED** [review]
LGTM. Thanks!

**Review by siriak — APPROVED** [review]
LGTM

**Pull request merged**